### PR TITLE
MAYA-127393 allow USD ref to be relative

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
- Refactor the Python helper to handle the relative-path UI in file dialogs.
- Make it more generic by abstracting to which file the path will be relative to, the label of the UI and the option var used.
- Add makePathRelativeTo function to UsdMayaUtilFileSystem to make any file path relative to a given directory.
- Add requireUsdPathsRelativeToEditTargetLayer function to verify if USD ref paths should be relative.
- Make the USD context menu use the new relative ref UI and functions.
- Refactor its code a bit to make reading the generated scripts easier.

### CODE REVIEW

File: lib/mayaUsd/utils/utilFileSystem.h [hunk 1]
```diff
@@ -34,6 +34,17 @@ std::string resolvePath(const std::string& filePath);
 MAYAUSD_CORE_PUBLIC
 std::string getDir(const std::string& fullFilePath);
 
+/*! \brief takes in two absolute file paths and returns the first one to second one.
+           Also return a boolean: true if the path was successfully made relative,
+           false otherwise.
+
+           If the second path is not absolute or is not reachable from the first,
+           then the returned path will still be absolute.
+ */
```
Comment (vlasovi) [inline lib/mayaUsd/utils/utilFileSystem.h:43]: We need to change slightly the behaviour of this function and document it, in order for it to correspond to behaviour of getPathRelativeToMayaSceneFile, where it is written: "When there is no scene file, the absolute (input) path will be returned."

**Review by vlasovi — CHANGES_REQUESTED** [review]
Behaviour of the function makePathRelativeTo has to be slightly changed, see the appropriate comments.

File: lib/mayaUsd/utils/utilFileSystem.cpp [hunk 2]
```diff
@@ -117,27 +117,39 @@ std::string UsdMayaUtilFileSystem::getMayaSceneFileDir()
...
 {
     ghc::filesystem::path absolutePath(fileName);
-    ghc::filesystem::path basePath(getMayaSceneFileDir());
 
-    // If Maya scene file doesn't exist yet, use the absolute path
-    if (basePath.empty()) {
-        return fileName;
+    // If Maya scene file doesn't exist yet, use the unchanged path
+    if (relativeToDir.empty()) {
+        return std::make_pair(fileName, false);
```
[the remark this answers is not in the document]
Comment (pierrebai-adsk) [reply lib/mayaUsd/utils/utilFileSystem.cpp:128]: Good catch< you only warned when teh scen path was valid but not reachable. Will fix it.

**Comment by pierrebai-adsk** [comment]
There is a conflict with dev. I will close this PR, rebase and re-open.

**Pull request closed without merging**